src/Features/CSharp/Portable/CodeFixes/Async/CSharpAddAwaitCodeFixProvider.cs

```diff
@@ -76,8 +72,19 @@ private bool IsCorrectReturnType(ExpressionSyntax expression, SemanticModel sema
 
         private static ExpressionSyntax ConvertToAwaitExpression(ExpressionSyntax expression)
         {
-            return SyntaxFactory.AwaitExpression(expression)
-                                .WithAdditionalAnnotations(Formatter.Annotation);
+            AwaitExpressionSyntax result;
+
+            if (expression.HasLeadingTrivia)
```

Comment:
@jmarolf There are other expression types like this. We need to figure them out first.
For CS4016 cases:
`return null /*0*/ ?? Task.FromResult(1) /*1*/`
`return null /*0*/ as Task<int> /*1*/`
 Can you come up the CS4014 and similar VB cases as well?